### PR TITLE
Add pre-commit and lint files

### PR DESCRIPTION
**Pull request by jbampton**
Add codespell, markdownlint, misspell and yamllint

https://pre-commit.com/
https://github.com/codespell-project/codespell
https://github.com/igorshubovych/markdownlint-cli
https://github.com/client9/misspell
https://yamllint.readthedocs.io/en/stable/

### CODE REVIEW

**Comment by jbampton** [comment]
GitHub Actions don't run the first time they are added to a repository.

**Comment by jbampton** [comment]
Also added `prettier` 

https://prettier.io/

**Comment by 1995parham** [comment]
Github action is awesome, it is great to have, I add one of the members to review it if he has time.

**Comment by smf8** [comment]
Thank you for your contribution. I'll review and merge it as soon as possible.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  misspell:
+    name: Check Spelling Misspell All Files In Commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install
+        run: wget -O - -q https://git.io/misspell | sh -s -- -b .
+      - name: Misspell
+        run: git ls-files --empty-directory | xargs ./misspell -error
+  pre-commit:
```
Comment (smf8) [inline .github/workflows/lint.yml:15]: pre commits are supposed to be `pre-commits` :D I don't think there should be any stage in github actions.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: fix-byte-order-marker
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+  # - repo: https://github.com/Lucas-C/pre-commit-hooks
+  #  rev: v1.1.10
+  #  hooks:
+  #    - id: remove-tabs
+  - repo: https://github.com/codespell-project/codespell
```
Comment (smf8) [inline .pre-commit-config.yaml:30]: We don't need this

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        name: Run codespell
+        description: Check spelling with codespell
+        entry: codespell --ignore-words=codespell.txt
+  - repo: https://github.com/pre-commit/mirrors-prettier
```
Comment (smf8) [inline .pre-commit-config.yaml:37]: Also this. We won't put code snippets directly in this repository.

**Comment by jbampton** [comment]
All the top projects use `pre-commit` like `Apache Airflow` and `mruby`.

Just shows how little you know :)

**Comment by smf8** [comment]
I for sure know **very little**. But I'm not comparing this repo with a large project like `Apache Airflow`. It's just a small archive for non-code content and I think it's just not necessary. If you feel like it, apply the changes or close the PR.

Thank you for your time.

**Pull request closed without merging**